datasets/ru/egrul/crawler.py

```diff
@@ -697,17 +698,17 @@ def crawl_index(context: Context, url: str) -> Set[str]:
     Returns:
         A set of ZIP archive URLs.
     """
-    archives: Set[str] = set()
-    doc = context.fetch_html(url)
-    for a in doc.findall(".//a"):
-        link_url = urljoin(url, a.get("href"))
-        if not link_url.startswith(url):
-            continue
-        if link_url.endswith(".zip"):
-            archives.add(link_url)
-            continue
-        archives.update(crawl_index(context, link_url))
-    return archives
+    with open(url, "r", encoding="utf-8") as f:
+        doc = html.fromstring(f.read())
+        doc.make_links_absolute(context.data_url)
```

Comment:
you're making the links absolute, relative to http://internal-data.opensanctions.org/ru_egrul/egrul.itsoft.ru/EGRUL_406/01.01.2022_FULL/

but you actually want to be making `fetch_internal_data()` calls using relative **paths** similar to the call for `ru_egrul/egrul.itsoft.ru/EGRUL_406/01.01.2022_FULL/index.html` e.g. `fetch_internal_data("ru_egrul/egrul.itsoft.ru/EGRUL_406/01.01.2022_FULL/EGRUL_FULL_2022-01-01_1.zip")`.